src/components/VmActions/index.js

```diff
@@ -80,6 +81,7 @@ const VmDropdownActions = ({ actions, id }) => {
             confirmation={action.confirmation}
             shortTitle={action.shortTitle}
             icon={action.icon}
+            onClick={action.onClick}
```

Comment:
I want to draw attention to this change. I think this was a bug, but I could also be missing something.

The onClick action wasn't being passed to the VM Drop Down Action buttons on the VM card view. Only the primary action kept its onClick property. This never caused an issue because all the other actions use a confirmation dialog which was being passed.

Reply:
no idea. possibly a bug indeed. @sjd78 ?

Reply:
I'm not sure I know what you mean on this one. Could you provide a bit more detail? 

Right now, MenuItemAction does this for onClick:

```javascript
onClick={() => {                                                         
  onClick && onClick()                                                   
  document.dispatchEvent(new MouseEvent('click'))                        
}
```

Reply:
Sure.  Here is the action definition for RDP console, which has the `onClick` definition:
https://github.com/oVirt/ovirt-web-ui/blob/09be1f1b36f5972e0efa6c41ca3128053872db58/src/components/VmActions/index.js#L162-L169

The onClick's `e.preventDefault()` will fail since, with the `MenuItemAction` `onClick` implementation you listed, nothing is passed in and e will be `undefined`.  The fix would be the have the `MenuItemAction` `onClick` implementation be:

```jsx
    <MenuItem
      onClick={e => {
        onClick && onClick(e)
        document.dispatchEvent(new MouseEvent('click'))
      }}
      {...rest}
    >
      <span>{shortTitle}</span> {icon}
    </MenuItem>
```

or to use the ... operators to handle all potential function arguments:
```jsx
    <MenuItem
      onClick={(...args) => {
        onClick && onClick(...args)
        document.dispatchEvent(new MouseEvent('click'))
      }}
      {...rest}
    >
      <span>{shortTitle}</span> {icon}
    </MenuItem>
```

Hope that helps.

You can either add that to this PR or spin off a new issue + PR just to handle the "Console RDP button doesn't work".

Reply:
Thanks so much for taking the time to explain this. I think I got it fixed.